packages/shared/components/TextHint.svelte

```diff
@@ -1,63 +1,85 @@
+    export let icon: IconEnum | undefined = undefined
     export let text: string = ''
-    export let textClasses: string = ''
-    export let textColor = 'gray-700'
-    export let textDarkColor = 'gray-400'
+    export let widthFull: boolean = false
 
-    let _classes: string
-    let _icon: string
-    let _iconClasses: string
+    $: iconClasses = DEFAULT_VALUES[variant].iconClasses
+    $: icon ||= DEFAULT_VALUES[variant].icon
 
-    function setClasses(): void {
-        if (primary) {
-            _classes = 'bg-gray-50 dark:bg-gray-800'
-        } else if (secondary) {
-            _classes = 'bg-gray-50 dark:bg-gray-800'
-        } else if (success) {
-            _classes = 'bg-green-50 dark:bg-green-500'
-            _icon = 'checkmark-filled'
-            _iconClasses = 'text-green-700 dark:text-green-700'
-        } else if (danger) {
-            _classes = 'bg-red-50 dark:bg-red-500'
-            _icon = 'error-filled'
-            _iconClasses = 'text-red-500 dark:text-red-500'
-        } else if (warning) {
-            _classes = 'bg-yellow-50 dark:bg-yellow-500'
-            _icon = 'exclamation-filled'
-            _iconClasses = 'text-yellow-700 dark:text-yellow-700'
-        } else if (info) {
-            _classes = 'bg-blue-50 dark:bg-blue-500'
-            _icon = 'info-filled'
-            _iconClasses = 'text-blue-600 dark:text-blue-600'
+    const DEFAULT_VALUES: {
+        [Variant in TextHintVariant]: {
+            icon: IconEnum | undefined
+            iconClasses: string | undefined
         }
+    } = {
+        [TextHintVariant.Primary]: {
+            icon: undefined,
+            iconClasses: undefined,
+        },
+        [TextHintVariant.Secondary]: {
+            icon: undefined,
+            iconClasses: undefined,
+        },
+        [TextHintVariant.Success]: {
+            icon: IconEnum.CheckmarkFilled,
+            iconClasses: 'text-green-700 dark:text-green-700',
+        },
+        [TextHintVariant.Danger]: {
+            icon: IconEnum.ErrorFilled,
+            iconClasses: 'text-red-500 dark:text-red-500',
+        },
+        [TextHintVariant.Warning]: {
+            icon: IconEnum.ExclamationFilled,
+            iconClasses: 'text-yellow-700 dark:text-yellow-700',
+        },
+        [TextHintVariant.Info]: {
+            icon: IconEnum.InfoFilled,
+            iconClasses: 'text-blue-600 dark:text-blue-600',
+        },
     }
-
-    $: $$props, setClasses()
 </script>
 
 {#if text}
-    <div class="flex flex-row items-center p-4 rounded-lg dark:bg-opacity-10 {_classes} {classes}">
-        <Icon
-            icon={icon ?? _icon}
-            height={iconSize}
-            width={iconSize}
-            primaryColor="white"
-            classes="mr-3 fill-current {_iconClasses} {iconClasses}"
-        />
-        <Text fontSize="14" lineHeight="5" color={textColor} darkColor={textDarkColor} classes={textClasses}>
+    <hint class="hint {variant}" class:w-full={widthFull}>
```

Comment:
I would put a clearer custom tag,maybe something like this:
```suggestion
    <text-hint class="{variant}" class:w-full={widthFull}>
```
and `.hint` is not necessary